File: examples/syntra-node/examples/retry-basic.ts
/**
 * retry-basic.ts — minimal usage demo for @ashhart/syntra-client.
 *
 * Run (after building):
 *   node --loader ts-node/esm examples/retry-basic.ts
 *
 * Or compile first:
 *   npm run build
 *   node dist/examples/retry-basic.js   # (copy example to src/ first)
 *
 * Requires a running Syntra instance and a capsule installed at CAPSULE_PATH.
 * Apache-2.0
 */

import { RetryClient } from "../src/index.js";

const SYNTRA_URL = process.env["SYNTRA_URL"] ?? "http://localhost:8787";
const ADMIN_KEY = process.env["SYNTRA_ADMIN_KEY"] ?? "dev-key";
const CAPSULE_PATH =
  process.env["SYNTRA_CAPSULE_PATH"] ??
  "/tenants/myteam/jobs/retry/capsules/router";
const TARGET_URL = process.env["TARGET_URL"] ?? "https://httpbin.org/get";

const client = new RetryClient({
  baseUrl: SYNTRA_URL,
  adminKey: ADMIN_KEY,
  capsulePath: CAPSULE_PATH,
  timeoutMs: 3000,
  onFeedbackError: (err) => {
    console.warn("Feedback error (non-fatal):", err);
  },
});

console.log(`Sending request to ${TARGET_URL} via RetryClient...`);

const response = await client.request("GET", TARGET_URL);

console.log(`Response status: ${response.status}`);

if (response.ok) {
  const body = await response.json();
  console.log("Response body (truncated):", JSON.stringify(body).slice(0, 200));
} else {
  console.warn("Request did not succeed with 2xx status.");
}

File: examples/syntra-rs/examples/retry_basic.rs
// Copyright 2026 Ash Hart. Apache-2.0.

//! Minimal end-to-end example for the Syntra retry client.
//!
//! Run a Syntra appliance at `SYNTRA_URL` (default `http://localhost:8787`)
//! with the demo retry-tuning capsule installed, then:
//!
//! ```text
//! SYNTRA_ADMIN_KEY=... cargo run --example retry_basic --release
//! ```

use std::env;

use syntra_client::retry::RetryClient;
use syntra_client::SyntraClient;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let url = env::var("SYNTRA_URL").unwrap_or_else(|_| "http://localhost:8787".into());
    let admin_key = env::var("SYNTRA_ADMIN_KEY").unwrap_or_default();
    let capsule = env::var("SYNTRA_CAPSULE_PATH")
        .unwrap_or_else(|_| "/tenants/demo/jobs/retry/capsules/router".into());
    let target = env::var("TARGET_URL").unwrap_or_else(|_| "https://httpbin.org/status/200".into());

    let syntra = SyntraClient::new(url, admin_key, capsule);
    let client = RetryClient::builder(syntra)
        .on_feedback_error(|e| eprintln!("feedback error: {e}"))
        .build();

    let resp = client.get(&target)?;
    println!("status: {}", resp.status());
    Ok(())
}
